### PR TITLE
feat: snippets

### PR DESCRIPTION
**Pull request by bayou-brogrammer**
## Goals

This PR adds snippets which is being discussed in #41 .

It will generate snippet lines for *ONLY* our containerfile module

- [x] allow for modules to specify snippets
- [x] append snippets to module run inside Containerfile
- [ ] write supporting test

## Preview
Recipe.yml

```yml
# image will be published to ghcr.io/<user>/<name>
name: orora

description: A starting point for further customization of uBlue images. Make your own! https://ublue.it/making-your-own/

base-image: ghcr.io/ublue-os/silverblue-main
image-version: latest # latest is also supported if you want new updates ASAP

modules:
  - type: signing
    snippets:
      - COPY --from=ghcr.io/blue-build/cli:latest-installer /out/BLUEBUILD /usr/bin/BLUEBUILD
```

Generated Container File snippet

```docker
RUN chmod +x /tmp/modules/signing/signing.sh && source /tmp/exports.sh && /tmp/modules/signing/signing.sh '{"type":"signing","snippets":["COPY --from=ghcr.io/blue-build/cli:latest-installer /out/BLUEBUILD /usr/bin/BLUEBUILD"]}'
COPY --from=ghcr.io/blue-build/cli:latest-installer /out/BLUEBUILD /usr/bin/BLUEBUILD
```

### CODE REVIEW

**Review by gmpinder — CHANGES_REQUESTED** [review]
Further discussions still need to be had for how to support this and for which modules:

https://github.com/blue-build/cli/issues/41

**Comment by xynydev** [comment]
I added my comment in #41 alongside some concerned questions on Discord, but here's more of a PR review:

- What's the usecase of adding snippets through the signing module? I understand this PR supports any module for snippets. Why, what's the benefit?
- Even if there is a benefit, it doesn't make sense to me to add another default key to the module configuration that *isn't related to the modules functionality at all*.
- This PR might be fine, though, if you refactor it to just work with the pre-existing `containerfile` module.

File: src/commands/template.rs [hunk 1]
```diff
@@ -121,12 +121,12 @@ fn has_cosign_file() -> bool {
 }
 
 #[must_use]
-fn get_containerfile_list(module: &Module) -> Option<Vec<String>> {
-    if module.module_type.as_ref()? == "containerfile" {
+fn get_module_type_list(module: &Module, typ: &str, list_key: &str) -> Option<Vec<String>> {
+    if module.module_type.as_ref()? == typ {
```
Comment (gmpinder) [inline src/commands/template.rs:125]:
```suggestion
fn get_module_type_list(module: &Module, type: &str, list_key: &str) -> Option<Vec<String>> {
    if module.module_type.as_ref()? == type {
```

File: .gitignore [hunk 2]
```diff
@@ -3,4 +3,5 @@
 .vscode/
 
 # Local testing for bluebuild recipe files
-/config/
+/config/*
+Containerfile
```
Comment (gmpinder) [inline .gitignore:7]: GitHub complaining about not ending the file with a newline.
```suggestion
Containerfile

```

**Review by gmpinder — CHANGES_REQUESTED** [review]

**Review by gmpinder — APPROVED** [review]

**Review by gmpinder — APPROVED** [review]

File: src/commands/template.rs [hunk 3]
```diff
@@ -121,12 +121,12 @@ fn has_cosign_file() -> bool {
 }
 
 #[must_use]
-fn get_containerfile_list(module: &Module) -> Option<Vec<String>> {
-    if module.module_type.as_ref()? == "containerfile" {
+fn get_module_type_list(module: &Module, type: &str, list_key: &str) -> Option<Vec<String>> {
+    if module.module_type.as_ref()? == type {
```
[the remark this answers is not in the document]
Comment (bayou-brogrammer) [reply src/commands/template.rs]: Yeah it's an unfortunate case in rust. Same with mod

**Pull request merged**